src/features/explicit-resource-management.md

```diff
@@ -0,0 +1,245 @@
+
+For example, when working with [`ReadableStreamDefaultReader`](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStreamDefaultReader), it's crucial to call `reader.releaseLock()` to unlock the stream and allow it to be used elsewhere. However, error handling introduces a common problem: if an error occurs during the reading process, and you forget to call `releaseLock()` before the error propagates, the stream remains locked. Let's start with a naive example:
+
+```javascript
+let responsePromise = null;
+
+async function readFile(url) {  
+    if (!responsePromise) {
+        // Only fetch if we don't have a promise yet
+        responsePromise = fetch(url);
+    }
+    const response = await responsePromise;
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+    const processedData = await processData(response);
+
+    // Do something with processedData
+    ...
+ }
+
+async function processData(response) {
+    const reader = response.body.getReader();
+    let done = false;
+    let value;
+    let processedData;
+    
+    while (!done) {
+        ({ done, value } = await reader.read());
+        if (value) {
+            // Process data and save the result in processedData
+            ...
+            // An error is thrown here!
+        }
+    }
+    
+    // Because the error is thrown before this line, the stream remains locked.
+    reader.releaseLock(); 
+
+    return processedData;
+  }
+ 
+ readFile('https://example.com/largefile.dat');
+```
+So it is crucial for developers to have `try...finally` block while using streams and put `reader.releaseLock()` in `finally`. This pattern ensures that `reader.releaseLock()` is always called.
+
+```javascript
+let responsePromise = null;
+
+async function readFile(url) {  
+    if (!responsePromise) {
+        // Only fetch if we don't have a promise yet
+        responsePromise = fetch(url);
+    }
+    const response = await responsePromise;
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+    const processedData = await processData(response);
+
+    // Do something with processedData
+    ...
+ }
```

Comment:
IMO you don't need to keep repeating the `readFile` function in subsequent examples if nothing in it changes.